src/org/jetbrains/java/decompiler/main/plugins/JarPluginLoader.java

```diff
@@ -0,0 +1,51 @@
+package org.jetbrains.java.decompiler.main.plugins;
+
+import java.io.*;
+import java.net.URI;
+import java.nio.file.*;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+// Finds plugins included in the main Quiltflower Jar via Jar-In-Jar
+public class JarPluginLoader {
+  static List<Class<?>> PLUGIN_CLASSES = new ArrayList<>();
+
+  public static void init() {
+    try {
+      File myFile = new File(JarPluginLoader.class.getProtectionDomain().getCodeSource().getLocation().toURI());
+
+      // Ensure we are running out of a file
+      if (myFile.exists() && !myFile.isDirectory() && myFile.getPath().endsWith(".jar")) {
+        URI uri = URI.create("jar:" + myFile.toURI());
+
+        Map<String, String> env = new HashMap<>();
+        // TODO: is this needed?
+        env.put("create", "true");
```

Comment:
Probably not since that flag just makes the FS provider create the file if missing, see the [`jdk.zipfs` module docs](https://docs.oracle.com/en/java/javase/19/docs/api/jdk.zipfs/module-summary.html)

Reply:
makes sense, ty!